lib/imagemounter/index.js

```diff
@@ -0,0 +1,119 @@
+const { exists, lstat } = fs;
+class ImageMounter extends BaseServicePlist {
+
+    _checkReturnError(ret) {
+        if (ret.Error) {
+            throw new Error(ret.Error);
+        }
+    }
+
+    /**
+     * Lookup for mounted images.
+     * @param {string?} imageType Type of image, `Developer` by default.
+     * @returns {Promise<Buffer[]>} Signature of each mounted image.
+     */
+    async lookup(imageType = 'Developer') {
+        const ret = await this._plistService.sendPlistAndReceive({
+            Command: 'LookupImage',
+            ImageType: imageType
+        });
+        this._checkReturnError(ret);
+        return ret.ImageSignature || [];
+    }
+
+    /**
+     * Check if developer image is mounted.
+     */
+    async isDeveloperImageMounted() {
+        return (await this.lookup()).length > 0;
+    }
+
+    /**
+     * Mount image for device.
+     * @param {string} imageFilePath The file path of image.
+     * @param {string} imageSignatureFilePath The signature file path of given `DeveloperDiskImage.dmg`
+     * @param {string?} imageType Type of image, `Developer` by default.
+     */
+    async mount(imageFilePath, imageSignatureFilePath, imageType = 'Developer') {
+        //check file stats
+        if (!await exists(imageFilePath)) {
+            throw new Error(`The image file provided not exists: ${imageFilePath}`);
+        }
+        const imageFileStat = await lstat(imageFilePath);
+        if (imageFileStat.isDirectory()) {
+            throw new Error(`The given image file is expected to be a file, a directory was given: ${imageFilePath}`);
+        }
+        if (!await exists(imageSignatureFilePath)) {
+            throw new Error(`The signature file provided not exists: ${imageFilePath}`);
+        }
+        if ((await lstat(imageSignatureFilePath)).isDirectory()) {
+            throw new Error(`The given signature file is expected to be a file, a directory was given: ${imageFilePath}`);
+        }
+        //read signature
+        const signature = await new B((resolve, reject) => {
+            readFile(imageSignatureFilePath, (err, data) => {
+                if (err != null) {
+                    reject(err);
+                } else {
+                    resolve(data);
+                }
+            });
+        });
+        const mountedImages = await this.lookup(imageType);
+        if (mountedImages.find((mountedSignature) => signature.equals(mountedSignature))) {
+            log.info(`An image with same signature of ${imageSignatureFilePath} is mounted. Doing nothing here`);
+            return;
+        }
+        //notify device
+        const imageSize = imageFileStat.size;
+        let ret = await this._plistService.sendPlistAndReceive({
+            Command: 'ReceiveBytes',
+            ImageSignature: signature,
+            ImageSize: imageSize,
+            ImageType: imageType
+        });
+        this._checkReturnError(ret);
+        if (ret.Status !== 'ReceiveBytesAck') {
```

Comment:
it'd be nice if we can define the type for `ret` somewhere. doesn't have to be in this PR.